core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1839,6 +1844,16 @@ abstract class RDD[T: ClassTag](
   def toJavaRDD() : JavaRDD[T] = {
     new JavaRDD(this)(elementClassTag)
   }
+
+  /**
+   * Whether the RDD is in a barrier stage. Spark must launch all the tasks at the same time for a
+   * barrier stage.
+   *
+   * An RDD is in a barrier stage, if at least one of its parent RDD(s), or itself, are mapped from
+   * a RDDBarrier. This function always returns false for a [[ShuffledRDD]], since a
+   * [[ShuffledRDD]] indicates start of a new stage.
+   */
+  def isBarrier(): Boolean = dependencies.exists(_.rdd.isBarrier())
```

Comment:
It doesn't cache the result. If we have a long linage, does it have performance penalty?